### PR TITLE
Add libgit2 0.27.9

### PR DESCRIPTION
**Pull request by intelligide**
Specify library name and version:  **Libgit2/0.27.9**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
Hi @intelligide !

Maybe you can re-use #286 It's almost approved and ready to be merged.

**Comment by intelligide** [comment]
Sorry, I hadn't seen this PR. I will modify my PR when libgit/0.28.3 is merged.

**Comment by danimtb** [comment]
@intelligide #286 has been merged. Please resolve conflicts here

**Comment by intelligide** [comment]
There is no error report.

File: recipes/libgit2/all/conanfile.py [hunk 1]
```diff
@@ -151,3 +151,5 @@ def package_info(self):
             self.cpp_info.system_libs.extend(["winhttp", "rpcrt4", "crypt32"])
         if self.settings.os == "Linux" and self.options.threadsafe:
             self.cpp_info.system_libs.append("pthread")
+        if self.settings.os == "Macos":
+            self.cpp_info.system_libs.append("curl")
```
Comment (uilianries) [inline recipes/libgit2/all/conanfile.py]: Hi! where did you find that libcurl is required for 0.27.9? If it is required, so we should use the libcurl conan package instead.

File: recipes/libgit2/config.yml [hunk 2]
```diff
@@ -1,3 +1,5 @@
 versions:
   "0.28.3":
     folder: all
+  "0.27.9":
+    folder: 0.27.x
```
Comment (jgsogo) [inline recipes/libgit2/config.yml]: If you specifically need `0.27.9`, it's ok; but please add 0.27.10 too as it is the last one of that series.

File: recipes/libgit2/0.27.x/conandata.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "0.27.9":
+    url: "https://github.com/libgit2/libgit2/archive/v0.27.9.tar.gz"
+    sha256: "adf17310b590e6e7618f070c742b5ee028aeeed2c60099bc4190c386b5060de1"
```
Comment (uilianries) [inline recipes/libgit2/0.27.x/conandata.yml]:
```suggestion
    sha256: "adf17310b590e6e7618f070c742b5ee028aeeed2c60099bc4190c386b5060de1"
  "0.27.10":
    url: "https://github.com/libgit2/libgit2/archive/v0.27.10.tar.gz"
    sha256: "f6fd26378ff71bd7a4b17b576c82c774a2e9c2d6b74b24718a8fb29551e1c4a5"
```

File: recipes/libgit2/config.yml (hunk 2 above)
Comment (uilianries) [inline recipes/libgit2/config.yml]:
```suggestion
    folder: 0.27.x
  "0.27.10":
    folder: 0.27.x
```

**Comment by intelligide** [comment]
@jgsogo 0.28 and 0.27 need differents recipes. 0.27 use curl and 0.28 use mbedtls.

**Pull request merged**